dev-tools/packaging/templates/docker/Dockerfile.tmpl

```diff
@@ -168,8 +168,10 @@ RUN for iter in {1..10}; do \
 done; \
 (exit $exit_code)
 {{- end }}
+USER 1000
+{{- else }}
+USER 0
```

Comment:
Without us having a test to ensure `runAsNonRoot` keeps working there is a chance someone accidentally breaks this in the future.

Are there tests for this not breaking in ECK that we can rely on? Or should be adding a regression test in Beats to make sure this keeps working?

We have a k8s test pipeline stage that could conceivably include this check eventually:

https://github.com/cmacknz/beats/blob/3806a9339c44d0e453f74760b2d21460405eac53/Jenkinsfile#L321-L332

Right now it just makes sure the default k8s YAML deploys successfully.

Reply:
Actually @cmacknz this is a catch-22. I can't get a test in place for this working until this feature is merged, and I know in what version of Beats this is functional.

Reply:
Got it, thanks